apps/dapp/src/components/Pages/Core/NewUI/DefendModal.tsx

```diff
@@ -0,0 +1,195 @@
+import { Popover } from 'components/Popover';
+import styled from 'styled-components';
+import { Button } from 'components/Button/Button';
+import { useWallet } from 'providers/WalletProvider';
+import { useEffect, useState } from 'react';
+import { formatTemple, getBigNumberFromString } from 'components/Vault/utils';
+import { ZERO } from 'utils/bigNumber';
+import { BigNumber } from 'ethers';
+import { CryptoSelector, CryptoValue, Input } from './HomeInput';
+import { TICKER_SYMBOL } from 'enums/ticker-symbol';
+import { formatToken } from 'utils/formatter';
+import env from 'constants/env';
+import { TempleStableAMMRouter__factory } from 'types/typechain';
+import { TreasuryIV__factory } from 'types/typechain/typechain';
+import { useNotification } from 'providers/NotificationProvider';
+import { useDebouncedCallback } from 'use-debounce';
+
+interface IProps {
+  isOpen: boolean;
+  onClose: () => void;
+}
+
+const { DAI, TEMPLE_TOKEN } = TICKER_SYMBOL;
+
+const inputConfig: CryptoValue = {
+  kind: 'value',
+  value: `${TEMPLE_TOKEN}`,
+};
+
+const outputConfig: CryptoSelector = {
+  kind: 'select',
+  cryptoOptions: [DAI],
+  onCryptoChange: () => {},
+  selected: DAI,
+};
+
+export const DefendModal: React.FC<IProps> = ({ isOpen, onClose }) => {
+  const { wallet, signer, balance } = useWallet();
+  const { openNotification } = useNotification();
+  const [inputValue, setInputValue] = useState('');
+  const [iv, setIv] = useState(ZERO);
+  const [quote, setQuote] = useState<BigNumber>();
+
+  // Set intrinsic value on mount
+  useEffect(() => {
+    const getIv = async () => {
+      if (!signer) return;
+      const treasuryIv = new TreasuryIV__factory(signer).attach(env.contracts.treasuryIv);
+      const { frax, temple } = await treasuryIv.intrinsicValueRatio();
+      setIv(frax.div(temple));
+    };
+    getIv();
+  }, [signer]);
+
+  // Fetch quote, debounced
+  const debouncedFetchQuote = useDebouncedCallback(async (amount: string) => {
+    const quote = await fetchQuote(inputValue);
+    setQuote(quote);
+  }, 750);
+
+  // Refresh quote on input change
+  useEffect(() => {
+    debouncedFetchQuote(inputValue);
+  }, [inputValue]);
+
+  // Get quote from router
+  const fetchQuote = async (input: string): Promise<BigNumber> => {
+    if (!signer || !wallet) return ZERO;
+    const router = new TempleStableAMMRouter__factory(signer).attach(env.contracts.templeV2Router);
+    const amountIn = getBigNumberFromString(input, 18);
+    if (amountIn.lte(ZERO)) return ZERO;
+    const { amountOut, priceBelowIV } = await router.swapExactTempleForStableQuote(
+      env.contracts.templeV2FraxPair,
+      amountIn
+    );
+    return priceBelowIV ? amountIn.mul(10_000).div(iv) : amountOut;
```

Comment:
`swapExactTempleForStableQuote()` already provides the 'best' quote, you don't need to manually check if it's above or below IV:

See:
https://github.com/TempleDAO/temple/blob/7e92dae66a78852b0bc24bae7c396dbe6c75ba8f/protocol/contracts/amm/TempleStableAMMRouter.sol#L215-L221

amountOut will automatically be set to either the defend rate or the swap rate.